### PR TITLE
doc: update docs of kusion concepts

### PR DESCRIPTION
**Pull request by liu-hm19**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

fix https://github.com/KusionStack/kusionstack.io/issues/525
#### 2. What is the scope of this PR (e.g. component or file name):

docs/kusion/3-concepts/
docs_versioned_docs/version-v0.12/3-concepts/

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

This PR adds the documentations to explain the concept of `Release`. 
#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: docs/kusion/3-concepts/6-spec.md [hunk 1]
```diff
@@ -20,3 +20,101 @@ Delivering an application to different environments with identical configuration
...
+        GVK: /v1, Kind=Service
+secretStore: null
+context: {}
+```
+
+From the example above, we can see that the `Spec` contains a list of `resources` required by the application. Each resource in the `Spec` needs to have `id`, `type`, `attributes`, `dependsOn`, and `extensions` fields: 
+
+- `id` is the unique key of this resource. An idiomatic way for Kubernetes resources is `apiVersion:kind:namespace:name`, and for Terraform resources is `providerNamespace:providerName:resourceType:resourceName`. 
+- `type` represents the type of runtime Kusion supports, and currently includes `Kubernetes` and `Terraform`. 
+- `dependsOn` contains all the other resources the resource depends on.
```
Comment (SparkYuan) [inline docs/kusion/3-concepts/6-spec.md:113]: please add an explanation about the field `attribute`

File: docs_versioned_docs/version-v0.12/3-concepts/4-workspace.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 id: workspace
-sidebar_label: Workspace
+sidebar_label: Workspaces
```
Comment (SparkYuan) [inline docs_versioned_docs/version-v0.12/3-concepts/4-workspace.md:3]: ?

File: docs/kusion/3-concepts/6-spec.md (hunk 1 above)
Comment (liu-hm19) [reply docs/kusion/3-concepts/6-spec.md:114]: OK, I've added a brief explanation about the field `attributes` in the doc.

**Review by SparkYuan — APPROVED** [review]
lgtm

**Pull request merged**